zboxcore/allocationchange/deletefile.go

```diff
@@ -35,18 +35,20 @@ func (ch *DeleteFileChange) ProcessChange(rootRef *fileref.Ref) error {
 			return errors.New("invalid_reference_path", "Invalid reference path from the blobber")
 		}
 	}
-	idx := -1
 	for i, child := range dirRef.Children {
+		if path == "/" {
+			dirRef.RemoveChild(i)
+			continue
+		}
 		if child.GetName() == ch.ObjectTree.GetName() && child.GetHash() == ch.ObjectTree.GetHash() {
-			idx = i
-			break
+			dirRef.RemoveChild(i)
+			rootRef.CalculateHash()
+			return nil
 		}
 	}
-	if idx < 0 {
+	if path != "/" {
```

Comment:
this `condition` seems is very strange, and difficult to understand? what is fixed by it?